File: Problems/1710_Maximum_Units_on_a_Truck.cpp
class Solution {
public:
    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
        sort(boxTypes.begin(), boxTypes.end(), [](auto& box1, auto& box2) {
            return box1[1] > box2[1];
        });
        int unitCount = 0;
        for(int i=0; i<boxTypes.size(); i++){
            if(boxTypes[i][0]<truckSize){
                unitCount += boxTypes[i][0]*boxTypes[i][1];
                truckSize -= boxTypes[i][0];
            }else{
                unitCount += truckSize*boxTypes[i][1];
                break;
            }
        }
        return unitCount;
    }
};

File: Problems/1710_Maximum_Units_on_a_Truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
        boxTypes.sort(reverse=True, key= lambda x: x[1])
        unitCount = 0
        for box in boxTypes:
            if truckSize > box[0]:
                truckSize -= box[0]
                unitCount += box[0]*box[1]
            else:
                unitCount += truckSize*box[1]
                break
        return unitCount